wordpress-mobile/WordPress-Android
# Handle viewpost deeplink in view model

This PR introduces 2 changes:
- moves handling of the `wordpress://viewpost` links to the viewmodel. They're now handled by the `ReaderLinkHandler` since they are opened in the Reader post detail. In addition to this I've changed the behaviour a little bit. Previously if the blog ID and post ID were not present/parsed, the app just finished. I've changed the behaviour to open default reader instead. I think this is better.
- Adds check whether the user is logged in before opening a link. If the user is not logged in, the URI is cached and the login is triggered for a result. Once we get the result, we retrigger the cached URL. This is a very generic approach and works for all the use cases. 

To test:
- Make sure you're logged in with a blog with remote ID X and post with remote ID Y
- open a `wordpress://viewpost?blogId=X&postId=Y` deeplink
- Notice the post detail is opened in the reader post detail
- Go back
- Notice you go to the main activity instead of closing the app

To test:
- Make sure you're logged out
- open a `wordpress://viewpost?blogId=X&postId=Y` deeplink
- Notice you're redirected to Login process
- Log in with a blog with remote ID X and post with remote ID Y
- Finish login
- Notice the post detail is opened in the reader post detail
- Go back
- Notice you go to the main activity instead of closing the app


To test:
- Make sure you're logged in
- open a `wordpress://viewpost` deeplink
- Notice the reader is opened

To test
- Test other links (stats, pages, read) without the user being logged in
- Notice the user is redirected to the desired link once login ends
- Notice the links work as expected when the user is logged in


## Regression Notes
1. Potential unintended areas of impact
- all the deeplinks

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- I've tested multiple flows and everything is unit tested

3. What automated tests I added (or what prevented me from doing so)
- unit tests

PR submission checklist:
